### ISSUE TITLE
[Bug] Can Intel provide the Jupyter Notebook for pedestrian-detection?

### ISSUE THREAD

**Issue by qin53027** [openvinotoolkit/openvino#6745]
Hi, 

Last week, I have installed OpenVINO from the Github:  https://github.com/openvinotoolkit/openvino

Also I have download the example of Jupyter Notebooks from https://github.com/openvinotoolkit/openvino_notebooks

Ran two examples, it looks very good! 

Now I want to evaluate the PeopleNet Model and Intel pedestrain-detection model as example of following:
https://github.com/openvinotoolkit/open_model_zoo/tree/2020.1/models/intel/pedestrian-detection-adas-0002

Do we have the relative Jupyter Notebook to use for this pedestrain-detection model?

If not, please provide me the suggestions how to utilize the OpenVINO to quickly train and inference using above model?

At same time, I am going to read the full OpenVINO user guide.

Thanks
Jimmy

**Comment by brmarkus** [comment]
You found the pre-trained model `pedestrian-detection-adas-0002` in the Open-Model-Zoo - start downloading it using the Model-Downloader from your OpenVINO installation.
I had installed OpenVINO 2021.4 on my machine (pre-built version, not build from source), so I download the model like this:

```
$> cd /opt/intel/openvino_2021/deployment_tools/tools/model_downloader
$> mkdir -p /tmp/model/
$> ./downloader.py --name pedestrian-detection-adas-0002 -o /tmp/model/
```
This will download the model in all available precisions and you will get something like this:

```
$> tree /tmp/model/
/tmp/model/
└── intel
    └── pedestrian-detection-adas-0002
        ├── FP16
        │   ├── pedestrian-detection-adas-0002.bin
        │   └── pedestrian-detection-adas-0002.xml
        ├── FP16-INT8
        │   ├── pedestrian-detection-adas-0002.bin
        │   └── pedestrian-detection-adas-0002.xml
        └── FP32
            ├── pedestrian-detection-adas-0002.bin
            └── pedestrian-detection-adas-0002.xml
```

Then I try to find which of the existing Open-Model-Zoo examples are referencing the model `pedestrian-detection-adas-0002` (in other words, trying to find out whether there are already existing examples showing how to use this model, because for every model in the Open-Model-Zoo there is an example showing how to use it):

```
$> cd /opt/intel/openvino_2021/deployment_tools/open_model_zoo/demos
$> grep -rnw . -e 'pedestrian-detection-adas'
```

For OpenVINO in version 2021.4 I was able to find these references:

```
./single_human_pose_estimation_demo/python/models.lst:5:pedestrian-detection-adas-????
./single_human_pose_estimation_demo/python/README.md:35:* pedestrian-detection-adas-0002
./object_detection_demo/cpp/models.lst:14:pedestrian-detection-adas-????
./object_detection_demo/cpp/README.md:91:  - pedestrian-detection-adas-0002
./object_detection_demo/python/models.lst:19:pedestrian-detection-adas-????
./object_detection_demo/python/README.md:85:  - pedestrian-detection-adas-0002
```

The demos `single_human_pose_estimation_demo` and `object_detection_demo` have a Python demo (where `object_detection_demo` is available as a C++ demo as well) - looks like you might be able to try them in Python directly (in a shell or in an IDE of your choice) and then later run them in a Jupyter notebook?

Can you give it a try - and report back if that is working for you?

**Comment by qin53027** [comment]
@brmarkus, thanks so much for your quick response.

Since I did OpenVINO build from source in my local,  I do not have the /opt/intel/..... dir. 

But I did downloaded the open_model_zoo: https://github.com/openvinotoolkit/open_model_zoo/tree/master/tools/downloader

Based on your good information, I am able to find the examples you mentioned and  I am going to try that to see shortly.

**Comment by qin53027** [comment]
Hi @brmarkus, 

Now I have downloaded the pedestrian-detection-adas-0002 into my /tmp/model/intel dir with FP16, FP16-IN8, FP32 suddir....
After I l read the README.md, I try to use following command to run test. Here are my questions:

1. while I can use: --model_od = /tmp/model/intel/pedestrian-detection-adas-0002/FP16/pedestrian-detection-adas-0002.xml
   but I do not have .xml file for --model_hpe:   Do I need this file for human pose estimator to this test or where to find that file?

3. For the input video file,  can this application support the .MP4 and H264 format as well?

2. Since I need to do the benchmark comparison with PeopleNet,  after the test run,  can I find the accuracy data such as nAP, Recall, etc. which is important to me?  If not, can you let me know I can find them through OPenVINO on pedestrian-detection-adas-0002?

Thanks so much
Jimmy 


example run command
==================
python3 single_human_pose_estimation_demo.py -d CPU --model_od <path_to_model>/mobilenet-ssd.xml --model_hpe <path_to_model>/single-human-pose-estimation-0001.xml --input <path_to_video>/back-passengers.avi

**Comment by qin53027** [comment]
The README.md file in my /home/jiande/program/open_model_zoo/demos/single_human_pose_estimation_demo/python, tells me that need to run converter.py. so I did:

python3 /home/jiande/program/open_model_zoo/tools/downloader/converter.py --list models.lst

But it tells me that 
No matching models: "/tmp/model/intel/pedestrian-detection-adas-0002/FP32"

But in that dir.  *.bin and *.xml files are there. Why it gives the error?

Can you help here?

Thanks
Jimmy


In models.list file     ----- I have download the following two models.
=============
 #This file can be used with the --list option of the model downloader.
/tmp/model/intel/pedestrian-detection-adas-0002/FP32
/tmp/model2/public/single-human-pose-estimation-0001/single-human-pose-estimation-0001/models

**Comment by qin53027** [comment]
Finally I am able to fix the model path problem and can run the test with following command after I have downloaded ingle-human-pose-estimation-0001 model.

The screen shows the video but no bundding box and pose on the person except that it prnts the summary only on FPS...  

Thanks
Jimmy


run command
===========
python3 single_human_pose_estimation_demo.py -d CPU --model_od /tmp/model/intel/pedestrian-detection-adas-0002/FP32/pedestrian-detection-adas-0002.xml  --model_hpe /home/jiande/program/open_model_zoo/demos/single_human_pose_estimation_demo/python/output/public/single-human-pose-estimation-0001/FP32/single-human-pose-estimation-0001.xml --input /home/jiande/Pictures/sample_720p.mp4

**Comment by qin53027** [comment]
Finally when I run the object_detection_demo.py, the video shows the bunding box

**Comment by brmarkus** [comment]
Can you show the command line which finally worked for you?
You originally asked for a Jupyter Notebook - do you think you can create a Jupyter Notebook on your own, by just using the the code of "single_human_pose_estimation_demo.py"?

**Comment by vladimir-dudnik** [comment]
@qin53027 @brmarkus the demo supports different object detection models and these models may have difference in class identification for detected person (this depends on training parameters, training dataset used and so on). Below is the list of supported models, taken from demo [description](https://github.com/openvinotoolkit/open_model_zoo/tree/master/demos/single_human_pose_estimation_demo/python):

* [mobilenet-ssd](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/mobilenet-ssd/README.md)
* [pedestrian-and-vehicle-detector-adas-0001](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/pedestrian-and-vehicle-detector-adas-0001/README.md)
* [pedestrian-detection-adas-0002](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/pedestrian-detection-adas-0002/README.md)
* [person-detection-retail-0013](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/person-detection-retail-0013/README.md)
* [person-vehicle-bike-detection-crossroad-0078](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/person-vehicle-bike-detection-crossroad-0078/README.md)
* [person-vehicle-bike-detection-crossroad-1016](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/person-vehicle-bike-detection-crossroad-1016/README.md)
* [ssd300](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/ssd300/README.md)
* [ssd512](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/ssd512/README.md)
* [ssd_mobilenet_v1_coco](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/ssd_mobilenet_v1_coco/README.md)
* [ssd_mobilenet_v1_fpn_coco](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/ssd_mobilenet_v1_fpn_coco/README.md)
* [ssd_mobilenet_v2_coco](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/ssd_mobilenet_v2_coco/README.md)
* [ssdlite_mobilenet_v2](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/ssdlite_mobilenet_v2/README.md)
* [single-human-pose-estimation-0001](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/single-human-pose-estimation-0001/README.md)

from which you may see, that for example, mobilenet-ssd model, which trained on PASCAL VOC dataset and mapping for class ids provided in file <omz_dir>/data/dataset_classes/[voc_20cl_bkgr.txt](https://github.com/openvinotoolkit/open_model_zoo/blob/master/data/dataset_classes/voc_20cl_bkgr.txt), where person has 16 as class id. In contrast to this, model pedestrian-detection-adas-0002 trained on internal dataset, but mapping to class ids provided by model description, namely, pedestrian has 1 as class id. To allow the flexibility of working with different models demo provide command line option `--person_label`, which you may use to inform demo what class id from model detections would correspond to person and demo will draw bounding boxes for such objects and also will run single-human-pose-estimation-0001 model on detected person.

**Comment by qin53027** [comment]
@brmarkus, following is the command which produce the correct result.  Usually I use the converter to covert the .py into Jupyter Notebook file. But since .py script works. It is fine to me.  Thanks

python3 object_detection_demo.py -d CPU -i /home/jiande/Pictures/people.mp4 -m /home/jiande/program/open_model_zoo/demos/object_detection_demo/python/output/public/ssd300/FP32/ssd300.xml -at ssd --labels /home/jiande/program/open_model_zoo/data/dataset_classes/voc_20cl_bkgr.txt

**Comment by qin53027** [comment]
@vladimir-dudnik, I try to run following command:

But it shows me the error
====================
single_human_pose_estimation_demo.py: error: unrecognized arguments: --labels /home/jiande/program/open_model_zoo/data/dataset_classes/voc_20cl_bkgr.txt

After I have removed --labels, it run except still can not see the bounding box on the person 

Do I need do something else?

Thanks
Jimmy


run command
===========
python3 single_human_pose_estimation_demo.py -d CPU  --model_od /tmp/model/intel/pedestrian-detection-adas-0002/FP32/pedestrian-detection-adas-0002.xml  --model_hpe /home/jiande/program/open_model_zoo/demos/single_human_pose_estimation_demo/python/output/public/single-human-pose-estimation-0001/FP32/single-human-pose-estimation-0001.xml --input /home/jiande/Pictures/people.mp4 --labels /home/jiande/program/open_model_zoo/data/dataset_classes/voc_20cl_bkgr.txt --person_label 16

**Comment by brmarkus** [comment]
The command line parameter is called `--person_label`. Can you try your command line again, please?

**Comment by qin53027** [comment]
If you see my command line, the last argument is --person_label 16

**Comment by brmarkus** [comment]
Please remove the parameter `--labels` from your command line.

The parameter `--person_label` expects a path and filename containing the classes and its mapping to IDs.

Depending on the used model (e.g. pedestrian detection, object-detection, pose-estimation) you need to provide another text-file. So for each supported model you would need to provide another text-file to the parameter `--person_label`.
